### PR TITLE
refactor(Popover): wrap popover with portal and popper.js

### PR DESCRIPTION
**Pull request by sun-tea**
- [ ] Feature
- [ ] Fix
- [x] Enhancement

## Brief
In order to fix our Select and Dropdown components, it seemed that a way was to use popper.js to help with the positioning. Since we already have a Popover component, I reworked this component to use popper.js and took the opportunity to add more utils like an EventListener component that can be instanciated each time we want to handle outside click.

Warning: if you test the story from the storybook, position of the arrow when you toggle `hasArrow` may not be accurate because it does not seem to update properly, but if you change the `placement` of the popover it should update.

## Requirements :

:warning: Requires running `yarn` command.

## Todo - Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.

### CODE REVIEW

File: src/Popover/elements.js [hunk 1]
```diff
@@ -1,34 +1,113 @@
...
+export const ARROW_SIZE = 8;
+export const CONTENT_PADDING = 1.8;
+export const MARGIN = 12;
 
-width: ${({ width }) => width};
+export const Arrow = styled.span`
+  color: ${({ theme: { palette } }) => palette.white};
 
-padding: 1.8rem;
+  text-shadow: ${({ placement }) =>
```
Comment (spartDev) [inline src/Popover/elements.js]: It's a bit redundant, move this part of code directly the right place, where you check for placement and add css

File: src/Popover/elements.js [hunk 2]
```diff
@@ -1,34 +1,113 @@
...
+      : '1px -5px 5px hsla(0, 0%, 0%, 0.1)'};
 
-border-radius: ${({ theme: { dimensions } }) => dimensions.radius};
+  position: absolute;
+  display: inline-block;
+  font-size: ${ARROW_SIZE * 2}px;
 
-box-shadow: 0 0.2rem 1.6rem 0 hsla(0, 0%, 0%, 0.1), 0 0.2rem 1.6rem 0 hsla(206, 23%, 69%, 0.1),
-  0 0 0 0.1rem hsl(207, 22%, 90%);
+  transform: ${({ placement }) =>
```
Comment (spartDev) [inline src/Popover/elements.js]: It's a bit redundant, move this part of code directly the right place, where you check for placement and add css

**Comment by sun-tea** [comment]
I broke the animation I'm trying to fix it

File: src/Popover/elements.js [hunk 3]
```diff
@@ -1,34 +1,113 @@
...
+      margin-left: 0;
+      margin-right: 0;
+    `}
+`;
+
+export const PopoverContentWrapper = styled.div`
+  background: ${({ theme: { palette } }) => palette.white};
+  border-radius: ${({ theme: { dimensions } }) => dimensions.radius};
+
+  box-shadow: 0 0.2rem 0.6rem 0 hsla(0, 0%, 0%, 0.1), 0 0.2rem 0.6rem 0 hsla(206, 23%, 69%, 0.1),
```
Comment (spartDev) [inline src/Popover/elements.js]: You should use constant for color (hsl(207, 22%, 90%)). You can check with @arthurmonnet to find the right reusable color for the shadow

File: src/ScrollBox/index.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    </Root>
+  );
+};
+
+/**
+ * PropTypes Validation.
+ */
+const { node, number } = PropTypes;
+Scrollbox.propTypes = {
+  children: node.isRequired,
```
Comment (spartDev) [inline src/ScrollBox/index.js]: You should put some documentation about props

File: src/ScrollBox/index.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  height: number,
```
Comment (spartDev) [inline src/ScrollBox/index.js]: You should put some documentation about props

File: src/Popover/index.js [hunk 6]
```diff
@@ -1,62 +1,243 @@
...
+            );
+          }}
+        </Reference>
+        {isOpen && (
+          <Portal>
+            <Popper modifiers={modifiers} placement={placement} positionFixed={positionFixed}>
+              {({ arrowProps, outOfBoundaries, placement, ref: popperRef, style }) => {
+                return (
+                  <PoseGroup flipMove={false}>
+                    <PopoverContentWrapperAnimation
```
Comment (spartDev) [inline src/Popover/index.js]: The animation here is never trigger

File: src/Popover/elements.js [hunk 7]
```diff
@@ -1,34 +1,113 @@
...
+    0 0 0 0.05rem hsl(207, 22%, 90%);
+  padding: ${CONTENT_PADDING}rem;
+  position: relative;
+
+  width: ${({ width }) => width};
 
-  height:14.14px;
-  width:14.14px;
+  ${props =>
+    props['data-placement'] === 'top' &&
```
Comment (Artikodin) [inline src/Popover/elements.js]: Does it relevant here to have the placement props passed as a data attribut ? If not, can you change `data-placement` to `placement` and use destructuring assignment as function parameter ?

File: src/Popover/elements.js [hunk 8]
```diff
@@ -1,34 +1,113 @@
...
+    css`
+      margin-bottom: ${MARGIN}px;
+    `}
 
-  transform: rotate(135deg);
+  ${props =>
+    props['data-placement'] === 'bottom' &&
```
Comment (Artikodin) [inline src/Popover/elements.js]: same

File: src/Popover/elements.js [hunk 9]
```diff
@@ -1,34 +1,113 @@
...
+    css`
+      margin-top: ${MARGIN}px;
+    `}
+
+  ${props =>
+    props['data-placement'] === 'left' &&
```
Comment (Artikodin) [inline src/Popover/elements.js]: same

File: src/Popover/elements.js [hunk 10]
```diff
@@ -1,34 +1,113 @@
...
+    css`
+      margin-right: ${MARGIN}px;
+    `}
+
+  ${props =>
+    props['data-placement'] === 'right' &&
```
Comment (Artikodin) [inline src/Popover/elements.js]: same

File: src/Popover/elements.js [hunk 11]
```diff
@@ -1,34 +1,113 @@
...
+    css`
+      margin-left: ${MARGIN}px;
+    `}
+
+  ${props =>
+    props['data-out-of-boundaries'] &&
```
Comment (Artikodin) [inline src/Popover/elements.js]: same

File: src/Popover/index.js (hunk 6 above)
Comment (sun-tea) [reply src/Popover/index.js]: Check my comment just above your review

File: src/Popover/elements.js (hunk 7 above)
Comment (sun-tea) [reply src/Popover/elements.js]: The name of the prop is because of the react-popper component

**Comment by sun-tea** [comment]
I broke the animation I'm trying to fix it

> Question: Have you seen in the [react-portal](https://www.npmjs.com/package/react-portal) there is a `<PortalWithState />` component witch handle when user click outside of the portal

As per https://github.com/tillersystems/Stardust/pull/615#issuecomment-508529482 I did not use it either

**Pull request merged**